File: gui/gui.py
import tkinter as tk

class GUI:
    def __init__(self):
        self.window = tk.Tk()
        self.window.title("GUI")

        self.init_button = tk.Button(self.window, text="Init", command=self.init_cmd)
        self.init_button.pack()

        self.run_button = tk.Button(self.window, text="Run", command=self.run_cmd)
        self.run_button.pack()

        self.stop_button = tk.Button(self.window, text="Stop", command=self.stop_cmd)
        self.stop_button.pack()

        self.status_button = tk.Button(self.window, text="Status", command=self.status_cmd)
        self.status_button.pack()

    def init_cmd(self):
        # initialization code here
        pass

    def run_cmd(self):
        # running code here
        pass

    def stop_cmd(self):
        # stopping code here
        pass

    def status_cmd(self):
        # status code here
        pass

    def run(self):
        self.window.mainloop()

if __name__ == "__main__":
    gui = GUI()
    gui.run()

File: gui/gui.swift
import Cocoa

class GUI: NSObject {
  let window: NSWindow
  let initButton: NSButton
  let runButton: NSButton
  let stopButton: NSButton
  let statusButton: NSButton

  override init() {
    window = NSWindow(contentRect: NSMakeRect(100, 100, 350, 70), styleMask: [.titled, .closable], backing: .buffered, defer: false)
    window.title = "GUI"

    initButton = NSButton(title: "Init", target: self, action: #selector(initCmd))
    runButton = NSButton(title: "Run", target: self, action: #selector(runCmd))
    stopButton = NSButton(title: "Stop", target: self, action: #selector(stopCmd))
    statusButton = NSButton(title: "Status", target: self, action: #selector(statusCmd))

    let vbox = NSView(frame: NSMakeRect(0, 0, 350, 70))
    vbox.addSubview(initButton)
    vbox.addSubview(runButton)
    vbox.addSubview(stopButton)
    vbox.addSubview(statusButton)

    window.contentView = vbox
    window.makeKeyAndOrderFront(nil)
  }

  @objc func initCmd() {
    // initialization code here
  }

  @objc func runCmd() {
    // running code here
  }

  @objc func stopCmd() {
    // stopping code here
  }

  @objc func statusCmd() {
    // status code here
  }
}

let gui = GUI()
NSApplication.shared.run()
